### PR TITLE
LF: Test evaluation order of exercise and exercise_by_key

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by remyhaemmerle-da** [comment]
together with #12411m #12471, and #12494, this fixes #11861.

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala [hunk 1]
```diff
@@ -455,6 +497,674 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
...
+            "key",
+            "maintainers",
+            "choice controllers",
+            "choice observers",
+          )
+        }
+      }
+
+      // TEST_EVIDENCE: Semantics: Evaluation order of exercise with output exceeding max nesting
+      "output exceeds max nesting" in {
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:797]:
```suggestion
      "result exceeds max nesting" in {
```
I think usually we call it choice result not output.

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala [hunk 2]
```diff
@@ -455,6 +497,674 @@ class EvaluationOrderTest extends AnyFreeSpec with Matchers with Inside {
...
+            gKey.templateId shouldBe T
+            msgs shouldBe Seq(
+              "starts test",
+              "maintainers",
+            )
+          }
+        }
+
+        // TEST_EVIDENCE: Semantics: Evaluation order of exercise_by_key of a cached global contract with failure authorization
+        "authorization failures" in {
```
Comment (cocreature) [inline daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:1051]: should we add visibility failures here as well?

**Review by cocreature — APPROVED** [review]
Great work, thank youk

File: daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:1051]: It was forgotten.

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/test/scala/com/digitalasset/daml/lf/speedy/EvaluationOrderTest.scala:1051]: added in fef1db52a0c28175482905a99701932c08ecb02f

**Pull request merged**